### ISSUE TITLE
Custom Rules Not Working

### ISSUE THREAD

**Issue by rajohns08** [realm/SwiftLint#794]
Xcode 7.3, SwiftLint 0.12.0

I'm trying the exact custom rule about ninjas from the README, but it doesn't seem to be working.

Here is my config file:

```yml
# Disable some rules that are turned on by default that we don't like
disabled_rules:
  - trailing_whitespace
  - file_length
  - force_cast
  - force_try
  - force_unwrapping
  - todo
  - type_body_length

# Customize existing rules to our liking
line_length:
  warning: 150
  error: 200

opt_in_rules:
  - vertical_whitespace

# Make our own rules
custom_rules:
  pirates_beat_ninjas: # rule identifier
    included: "*.swift" # regex that defines paths to include during linting. optional.
    name: "Pirates Beat Ninjas" # rule name. optional.
    regex: "([n,N]inja)" # matching pattern
    match_kinds: # SyntaxKinds to match. optional.
      - comment
      - identifier
    message: "Pirates are better than ninjas." # violation message. optional.
    severity: error # violation severity. optional.
  no_hiding_in_strings:
    regex: "([n,N]inja)"
    match_kinds: string
```
Here is my config output:
```
$ swiftlint rules --config .swiftlint.yml
Loading configuration from '.swiftlint.yml'
Invalid configuration for 'custom_rules'. Falling back to default.
+--------------------------------+--------+-------------+------------------------+--------------------------------------------------+
| identifier                     | opt-in | correctable | enabled in your config | configuration                                    |
+--------------------------------+--------+-------------+------------------------+--------------------------------------------------+
| closing_brace                  | no     | yes         | yes                    | warning                                          |
| colon                          | no     | yes         | yes                    | flexible_right_spacing: false                    |
| comma                          | no     | yes         | yes                    | warning                                          |
| conditional_returns_on_newline | yes    | no          | no                     | N/A                                              |
| control_statement              | no     | no          | yes                    | warning                                          |
| custom_rules                   | no     | no          | yes                    | user-defined                                     |
| cyclomatic_complexity          | no     | no          | yes                    | warning: 10, error: 20                           |
| empty_count                    | yes    | no          | no                     | error                                            |
| file_length                    | no     | no          | no                     | warning: 400, error: 1000                        |
| force_cast                     | no     | no          | no                     | error                                            |
| force_try                      | no     | no          | no                     | error                                            |
| force_unwrapping               | yes    | no          | no                     | warning                                          |
| function_body_length           | no     | no          | yes                    | warning: 40, error: 100                          |
| function_parameter_count       | no     | no          | yes                    | warning: 5, error: 8                             |
| leading_whitespace             | no     | yes         | yes                    | warning                                          |
| legacy_cggeometry_functions    | no     | yes         | yes                    | warning                                          |
| legacy_constant                | no     | yes         | yes                    | warning                                          |
| legacy_constructor             | no     | yes         | yes                    | warning                                          |
| legacy_nsgeometry_functions    | no     | yes         | yes                    | warning                                          |
| line_length                    | no     | no          | yes                    | warning: 150, error: 200                         |
| mark                           | no     | no          | yes                    | warning                                          |
| missing_docs                   | yes    | no          | no                     | warning: source.lang.swift.accessibility.public  |
| nesting                        | no     | no          | yes                    | warning                                          |
| opening_brace                  | no     | yes         | yes                    | warning                                          |
| operator_whitespace            | no     | no          | yes                    | warning                                          |
| private_outlet                 | yes    | no          | no                     | warning                                          |
| private_unit_test              | no     | no          | yes                    | warning: XCTestCase                              |
| return_arrow_whitespace        | no     | yes         | yes                    | warning                                          |
| statement_position             | no     | yes         | yes                    | (statement_mode) default, (severity) warning     |
| todo                           | no     | no          | no                     | warning                                          |
| trailing_newline               | no     | yes         | yes                    | warning                                          |
| trailing_semicolon             | no     | yes         | yes                    | warning                                          |
| trailing_whitespace            | no     | yes         | no                     | warning, ignores_empty_lines: false              |
| type_body_length               | no     | no          | no                     | warning: 200, error: 350                         |
| type_name                      | no     | no          | yes                    | (min_length) w/e: 3/0, (max_length) w/e: 40/1000 |
| valid_docs                     | no     | no          | yes                    | warning                                          |
| variable_name                  | no     | no          | yes                    | (min_length) w/e: 3/2, (max_length) w/e: 40/60   |
| vertical_whitespace            | yes    | yes         | yes                    | warning                                          |
+--------------------------------+--------+-------------+------------------------+--------------------------------------------------+
```

**Comment by norio-nomura** [comment]
Thank you for reporting.
> Invalid configuration for 'custom_rules'. Falling back to default.

It seems custom rule example in README.md was incorrect.

**Comment by rajohns08** [comment]
Do you have any example of one that should work?

**Comment by rajohns08** [comment]
`included` should have been `include` for `*.swift`

**Comment by norio-nomura** [comment]
Yeah, I fixed README.md in #795 and now it works.
You can find another examples from closed [issues](https://github.com/realm/SwiftLint/issues?q=is%3Aissue+custom_rules+is%3Aclosed).

**Comment by norio-nomura** [comment]
Oh, my commit closed this issue.
If you have any question, feel free to reopen or create new issue.

**Comment by norio-nomura** [comment]
Sorry, my bad.
#795 was incorrect.
`included` is correct keyword. `*.swift` should be `.*.swift`.
